sxg_rs/src/acme/mod.rs

```diff
@@ -12,4 +12,521 @@
+        let (fetcher, mut server) = crate::fetcher::mock_fetcher::create();
+        let public_key = EcPublicKey {
+            kty: "EC".to_string(),
+            crv: "P-256".to_string(),
+            x: vec![1],
+            y: vec![2],
+        };
+        // The client is handled by the code in `sxg_rs::acme`.
+        let client_thread = async {
+            let signer = crate::signature::mock_signer::MockSigner;
+            let ongoing_certificate_request = create_request_and_get_challenge_answer(
+                "https://acme.server/",
+                "admin@example.com",
+                "example.com",
+                public_key,
+                "csr content".to_string().into_bytes(),
+                fetcher,
+                signer,
+            )
+            .await
+            .unwrap();
+            assert_eq!(&ongoing_certificate_request.challenge_answer, "0HORFRxrqEtAB-vUh9iSnFBHE66qWX4bbU1SBWxOr5o.CmzeuaSxxfG8gIKRU_AgBzPa16nTt0H64JD7q1sZUUY");
+            let certificate_pem =
+                continue_challenge_validation_and_get_certificate(ongoing_certificate_request)
+                    .await
+                    .unwrap();
+            assert_eq!(certificate_pem, "content of certificate");
+        };
+        // The server side is mocked by directly putting all HTTP requests and responses.
+        // For each pair (`req`, `res`) variables in the following code block, we expect the client
+        // thread to fetch `req`, and we will return `res` to client side.
+        let server_thread = async {
+            let signer = crate::signature::mock_signer::MockSigner;
+
+            let req = HttpRequest {
+                body: vec![],
+                method: Method::Get,
+                headers: vec![],
+                url: "https://acme.server/".to_string(),
+            };
+            let res = HttpResponse {
+                status: 200,
+                headers: vec![],
+                body: r#"{
+                    "keyChange": "https://acme.server/key-change",
+                    "newAccount": "https://acme.server/new-acct",
+                    "newNonce": "https://acme.server/new-nonce",
+                    "newOrder": "https://acme.server/new-order",
+                    "revokeCert": "https://acme.server/revoke-cert"
+                }"#
+                .to_string()
+                .into_bytes(),
+            };
+            server.handle_next_request(req, res).await.unwrap();
+
+            let req = HttpRequest {
+                body: vec![],
+                method: Method::Get,
+                headers: vec![],
+                url: "https://acme.server/new-nonce".to_string(),
+            };
+            let res = HttpResponse {
+                status: 200,
+                headers: vec![("Replay-Nonce".to_string(), "1".to_string())],
+                body: vec![],
+            };
+            server.handle_next_request(req, res).await.unwrap();
+
+            let req = HttpRequest {
```

Comment:
Please add a test that exercises external account binding (see: https://docs.google.com/document/d/1hUTCIVOwjplMhPngKcerv2VSRnxJ1IaUqB-uP7W75DQ/edit#heading=h.x9snb54sjlu9)

Reply:
The EAB is now added as a field in `NewAccountRequestPayload`. The [EAB field](https://datatracker.ietf.org/doc/html/rfc8555#section-7.3.4) uses `{protected: ..., payload: ..., signature: ...}`, which has been tested in `jws.rs`.